app/src/main/java/com/example/ana/exampleapp/TestActivity.java

```diff
@@ -160,9 +171,57 @@ protected void onCreate(Bundle savedInstanceState) {
             r5.updateColor();
             r6.updateColor();
         }
+
+        allowLocation = settings.getBoolean("Location_enabled", true);
+        if (allowLocation) {
+            //if api > 23 then the user must grant the permission to access the network or location
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) != PackageManager.PERMISSION_GRANTED && ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION) != PackageManager.PERMISSION_GRANTED) {
```

Comment:
This line is too long and it is difficult to read, it would be great if you could short it (by assign to several boolean variables the conditions for example).